File: business/user/impl/src/main/java/com/user/login/viewmodel/LoginViewModel.java
package com.user.login.viewmodel;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;

import com.agri.pest.client.api.ServiceCode;
import com.agri.pest.client.model.request.PostCreateRequest;
import com.agri.pest.client.model.response.ResultAuthResponse;
import com.agri.pest.client.model.response.ResultUserProfileDto;
import com.common.base.BaseViewModel;
import com.user.login.data.LoginRepository;
import com.user.profile.data.Repository;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.rxjava3.disposables.Disposable;
import io.reactivex.rxjava3.schedulers.Schedulers;

public class LoginViewModel extends BaseViewModel {
    public MutableLiveData<ResultAuthResponse> loginResultLiveData = new MutableLiveData<ResultAuthResponse>();

    public MutableLiveData<ResultAuthResponse> registerResultLiveData = new MutableLiveData<ResultAuthResponse>();

    public MutableLiveData<String> errorLiveData = new MutableLiveData<String>();

    private MutableLiveData<String> toastMsg = new MutableLiveData<>("");

    private MutableLiveData<Boolean> agreeChecked = new MutableLiveData<>(false);

    private final LoginRepository repository = new LoginRepository();
    private final Repository profileRepository = new Repository();

    private final MutableLiveData<ResultUserProfileDto> userProfileLiveData = new MutableLiveData<>();


    public void login(String usernameOrPhone, String password) {
        String phoneError = repository.validatePhone(usernameOrPhone);
        if (phoneError != null) {
            toastMsg.setValue(phoneError);
            return;
        }
        String passwordError = repository.validatePassword(password);
        if (passwordError != null) {
            toastMsg.setValue(passwordError);
            return;
        }
        if (!(agreeChecked.getValue())) {
            toastMsg.setValue("请同意用户协议和隐私政策");
            return;
        }
        Disposable disposable = repository.login(usernameOrPhone, password)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(response -> {
                    if (response.getCode() == ServiceCode.SUCCESS) {
                        repository.saveUserInfo(response.getData());
                        repository.savePassword(password);
                        toastMsg.setValue("登录成功");
                    } else {
                        toastMsg.setValue("登录失败" + repository.errorCode(response.getCode()));
                    }
                    loginResultLiveData.setValue(response);
                }, error -> {
                    errorLiveData.setValue("登录失败:" + error.getMessage());
                });
        addDisposable(disposable);
    }

    public void register(String phone, String password, String username) {
        String phoneError = repository.validatePhone(phone);
        if (phoneError != null) {
            toastMsg.setValue(phoneError);
            return;
        }
        Disposable disposable = repository.register(phone, password, username)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(response -> {
                    if (response.getCode() == ServiceCode.SUCCESS) {
                        repository.saveUserInfo(response.getData());
                    } else {
                        toastMsg.setValue("注册失败" + repository.errorCode(response.getCode()));
                    }
                    registerResultLiveData.setValue(response);
                }, error -> {
                    errorLiveData.setValue("注册失败:" + error.getMessage());
                });
        addDisposable(disposable);
    }


    public MutableLiveData<ResultAuthResponse> getLoginResultLiveData() {
        return loginResultLiveData;
    }

    public MutableLiveData<ResultAuthResponse> getRegisterResultLiveData() {
        return registerResultLiveData;
    }

    public MutableLiveData<String> getErrorLiveData() {
        return errorLiveData;
    }

    public MutableLiveData<String> getToastMsg() {
        return toastMsg;
    }

    public void setAgreeChecked(Boolean checked) {
        agreeChecked.setValue(checked);
    }

    public MutableLiveData<ResultUserProfileDto> getUserProfileLiveData() {
        return userProfileLiveData;
    }

    public void getUserMes() {
        Disposable disposable = profileRepository.getUserMes()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(
                        response -> {
                            if (response.getCode() == ServiceCode.SUCCESS && response.getData() != null) {
                                userProfileLiveData.setValue(response);
                            } else {
                                userProfileLiveData.setValue(null);
                            }
                        },
                        error -> {
                            userProfileLiveData.setValue(null);
                        }
                );
        addDisposable(disposable);
    }
}

File: business/user/impl/src/main/java/com/user/LoginViewModel.kt
package com.user

import androidx.lifecycle.MutableLiveData
import com.common.base.BaseViewModel
import com.network.NetworkManager
import com.agri.pest.client.api.ServiceCode
import com.agri.pest.client.model.request.LoginRequest
import com.agri.pest.client.model.request.RegisterRequest
import com.agri.pest.client.model.response.ResultAuthResponse
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.schedulers.Schedulers

/**
 * 登录注册 ViewModel
 * 使用 AgriPest-Android-SDK 重构
 */
class LoginViewModel : BaseViewModel() {

    // 登录结果状态，使用 SDK 的数据模型
    val loginResultLiveData = MutableLiveData<ResultAuthResponse>()
    val registerResultLiveData = MutableLiveData<ResultAuthResponse>()
    val errorLiveData = MutableLiveData<String>()

    /**
     * 登录方法
     */
    fun login(usernameOrPhone: String, password: String) {
        val request = LoginRequest(
            usernameOrPhone = usernameOrPhone,
            password = password
        )
        
        val disposable = NetworkManager.api.login(request)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe({ response ->
                if (response.code == ServiceCode.SUCCESS && response.data != null) {
                    // 登录成功，更新 Token
                    response.data!!.token?.let { NetworkManager.setToken(it) }
                }
                loginResultLiveData.value = response
            }, { error ->
                errorLiveData.value = "登录请求失败: ${error.message}"
            })
        addDisposable(disposable)
    }

    /**
     * 注册方法
     */
    fun register(phone: String, password: String, username: String? = null) {
        val request = RegisterRequest(
            phone = phone,
            password = password,
            username = username
        )
        
        val disposable = NetworkManager.api.register(request)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe({ response ->
                if (response.code == ServiceCode.SUCCESS && response.data != null) {
                    // 注册成功，更新 Token
                    response.data?.token?.let { NetworkManager.setToken(it) }
                }
                registerResultLiveData.value = response
            }, { error ->
                errorLiveData.value = "注册请求失败: ${error.message}"
            })
        addDisposable(disposable)
    }
}
